### ISSUE TITLE
E1111 false positive with pathlib.Path.readlink()

### ISSUE THREAD

**Issue by rompe** [PyCQA/pylint#11114]
### Bug description

When assigning the result of `readlink()` to a variable, Pylint logs a false E1111 (assignment-from-no-return) positive. In reality, `readlink()` returns a `pathlib.Path` object.

`pylint-E1111-false-positive.py` :
```python
import pathlib

path = pathlib.Path("test.txt")
result = path.readlink()
```

### Configuration

```ini

```

### Command used

```shell
pylint --disable all --enable E1111 pylint-E1111-false-positive.py
```

### Pylint output

```python
************* Module pylint-E1111-false-positive
pylint-E1111-false-positive.py:4:0: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

### Expected behavior

No warning would be expected.

### Pylint version

```shell
pylint 4.0.6
astroid 4.0.4
Python 3.14.0rc3 (main, Sep 18 2025, 19:45:57) [MSC v.1944 64 bit (AMD64)]
```

### OS / Environment

- RedHat 9 with Python 3.14.6 
- Windows 11 with Python 3.14.0rc3

### Additional dependencies

```python

```

**Comment by clavedeluna** [comment]
I reproduced this with `uvx --python 3.14 --with astroid==4.0.4 --from pylint==4.0.6 \
    pylint --disable all --enable E1111 pylint-E1111-false-positive.py`

It doesn't happen in python 3.12 or 3.13, so it's an astroid inference bug on 3.14 only.
